shell/platform/android/io/flutter/embedding/engine/renderer/FlutterRenderer.java

```diff
@@ -360,5 +382,104 @@ public void dispatchSemanticsAction(
     public int systemGestureInsetRight = 0;
     public int systemGestureInsetBottom = 0;
     public int systemGestureInsetLeft = 0;
+    public List<DisplayFeature> displayFeatures = new ArrayList<DisplayFeature>();
+  }
+
+  /**
+   * Area of the viewport obstructed by a feature, eg. a hinge, a fold crease or camera cutout
+   *
+   * <p>Similar to {@link androidx.window.DisplayFeature}, used for reporting Fold / Hinge areas.
+   * The reason we define our own model is that we also want to support cutouts.
```

Comment:
Pitch this as a more comprehensive class that handles hinges/folds/cutouts rather than a justification for existence.